tests/framework/framework.go

```diff
@@ -38,10 +38,17 @@ var (
 	master       *string
 )
 
+type Config struct {
+	// Whether to skip creating a namespace. Use this ONLY for tests that do not require
+	// a namespace at all, like basic sanity or other global tests.
+	SkipNamespaceCreation bool
+}
+
 // Framework supports common operations used by functional/e2e tests. It holds the k8s and cdi clients,
 // a generated unique namespace, run-time flags, and more fields will be added over time as cdi e2e
 // evolves. Global BeforeEach and AfterEach are called in the Framework constructor.
 type Framework struct {
+	config *Config
```

Comment:
@awels did you try _embedding_ Config into Framework -- sort of an anonymous include? A reason to embed would be that you can reference any of the config fields simply by their name. Compare:
```
skipNs := f.SkipNamespaceCreation  // with embedding
```
vs.
```
skipNs := f.Config.SkipNamespaceCreation  // current code

```
It's not a big deal but is a nice feature of Go